source/agora/node/BlockStorage.d

```diff
@@ -194,24 +208,30 @@ public class BlockStorage : IBlockStorage
 
         Load the blockchain from the storage
 
+        Performs loading of the index and the last batch of blocks from disk.
+
+        Returns:
+            `false` when it fails to load.
+
     ***************************************************************************/
 
-    public void load ()
+    public bool load ()
     {
-        scope (failure) assert(0);
-
-        if (!this.path.exists)
-            mkdirRecurse(this.path);
+        try if (!this.root_path.exists)
+            mkdirRecurse(this.root_path);
+        catch (Exception e)
+            return false;
 
         if (!this.loadAllIndexes())
-            assert(0);
+            return false;
 
         // Add Genesis if the storage is empty
         if (this.height_idx.length == 0)
         {
             if (!this.saveBlock(GenesisBlock))
-                assert(0);
+                return false;
```

Comment:
This isn't about loading, but saving. So I think it should stay as `assert(0);`

Reply:
Although I see you do the same in the calling code.. It's a bit of an interesting case..

Reply:
Ok based on the commit message it makes sense, let it be like this.